packages/configurationmap/components/HashMap.tsx

```diff
@@ -0,0 +1,152 @@
+   */
+  hash: { [key: string]: React.ReactNode };
+  /**
+   * Priority of the heading that the `headline` prop renders in. Numbers map to <h1> through <h6>
+   */
+  headingLevel?: headingLevel;
+  /**
+   * The heading that labels the data
+   */
+  headline?: React.ReactNode;
+  /**
+   * Optional object with keys consisting of keys in the `hash` prop to be replaced, and with corresponding values of the replacement to be rendered.
+   */
+  renderKeys?: { [key: string]: string };
+}
+
+class HashMap extends React.PureComponent<HashMapProps, {}> {
+  public static defaultProps: Partial<HashMapProps> = {
+    headingLevel: 1,
+    emptyValue: "—"
+  };
+
+  constructor(props) {
+    super(props);
+
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getHeadline() {
+    const { headline, headingLevel } = this.props;
+    if (!headline) {
+      return null;
+    }
+
+    // Wrap in headline element and classes
+    return (
+      <ConfigurationMapHeading headingLevel={headingLevel}>
+        {headline}
+      </ConfigurationMapHeading>
+    );
+  }
+
+  getHashMap(headline, hashMap) {
+    // Increase the heading level for each nested description list,
+    // ensuring we don't surpass heading level <h6/>.
+    const headingLevel = this.props.headingLevel || 0;
+    const nextHeadingLevel = Math.min(headingLevel + 1, 6) as headingLevel;
+
+    return (
+      <HashMap
+        {...this.props}
+        hash={hashMap}
+        headingLevel={nextHeadingLevel}
+        key={`hash-map-${headline}`}
+        headline={headline}
+      />
+    );
+  }
+
+  getItemRow(headline, value) {
+    return (
+      <ConfigurationMapRow key={`hash-map-value-${headline}`}>
+        <ConfigurationMapLabel>{headline}</ConfigurationMapLabel>
+        <ConfigurationMapValue>{this.formatValue(value)}</ConfigurationMapValue>
+      </ConfigurationMapRow>
+    );
+  }
+
+  getItems() {
+    const { hash, renderKeys } = this.props;
+
+    return Object.keys(hash).map(key => {
+      const value = hash[key];
+
+      if (this.isHashMap(value)) {
+        return this.getHashMap(key, value);
+      }
+
+      if (renderKeys && Object.prototype.hasOwnProperty.call(renderKeys, key)) {
+        key = renderKeys[key];
+      }
+
+      return this.getItemRow(key, value);
+    });
+  }
+
+  render() {
+    const { hash } = this.props;
+
+    if (!hash || Object.keys(hash).length === 0) {
+      return null;
+    }
+
+    return (
+      <ConfigurationMapSection>
+        {this.getHeadline()}
+        {this.getItems()}
+      </ConfigurationMapSection>
+    );
+  }
+
+  formatValue(value) {
+    if (typeof value === "boolean") {
+      value = value.toString();
+    }
+
+    if (Array.isArray(value)) {
+      value = value.join(", ");
+    }
+
+    if (!value && this.props.emptyValue) {
+      value = this.props.emptyValue;
+    }
+
+    return value;
+  }
+
+  isHashMap(value) {
```

Comment:
sidenote: this "method" does not use `this` and thus does not need to be a method.

no action item, please just close this comment.